packages/story-editor/src/components/canvas/useInsertElement.js

```diff
@@ -56,14 +59,45 @@ function useInsertElement() {
    */
   const backfillResource = useCallback(
     (resource) => {
-      const { type, src, id, posterId, local } = resource;
+      const { type, src, id, posterId, local, baseColor, isExternal } =
+        resource;
+
+      if (local || !id || isExternal) {
+        return;
+      }
 
       // Generate video poster if one not set.
-      if (['video', 'gif'].includes(type) && id && !posterId && !local) {
+      if (['video', 'gif'].includes(type) && !posterId) {
         uploadVideoPoster(id, src);
       }
+      // Backfill base color, if one not set.
+      if (!Array.isArray(baseColor) || !baseColor.length) {
+        updateBaseColor({ resource });
+      }
+    },
+    [uploadVideoPoster, updateBaseColor]
+  );
+
+  const generateBaseColorExternal = useCallback(
```

Comment:
Why is this limited to external items only?

There could be a scenario where a regular media item doesn't have a baseColor, so this would be very useful to run for those cases as well.